wallet/src/main/java/com/android/identity_credential/wallet/presentation/OpenID4VPPresentationActivity.kt

```diff
@@ -490,15 +496,7 @@ class OpenID4VPPresentationActivity : FragmentActivity() {
             inputDescriptorObj["id"]!!.toString().run { substring(1, this.length - 1) }
         } else {
             try {
-                var vct = ""
-                val constraints = inputDescriptorObj["constraints"]!!.jsonObject
-                for (field in constraints["fields"]!!.jsonArray) {
-                    if (field.jsonObject["path"]!!.jsonArray[0].toString() == "\"\$.vct\"") {
-                        val vctField = field.jsonObject
-                        val filter = vctField["filter"]!!.jsonObject
-                        vct = filter["const"]!!.toString().run { substring(1, this.length - 1) }
-                    }
-                }
+                val (vct, _) = parseInputDescriptorForVc(inputDescriptorObj)
                 vct
             } catch (e: NullPointerException) {
                 Logger.d(TAG, "Error: Could not find const filter field: ${e.message}")
```

Comment:
This error msg might not be entirely accurate anymore since there may have been a different source of the error elsewhere in `parseInputDescriptorForVc`

Reply:
Updated. Also cleaned up some duplicate code.